examples/apps/color_finder.py

```diff
@@ -0,0 +1,659 @@
+
+    def on_finding_a_blob(self, blob_info):
+        ''' Determines whether Cozmo should continue to look at the blob, or drive towards it.
+            
+            Args:
+                blob_info (int, int): coordinates of the blob's center in self.pixel_matrix
+        '''
+        self.robot.set_center_backpack_lights(map_color_to_light[self.color_to_find])
+
+        x, y = blob_info
+        WM = 10.0*32/self.downsize_width  # WM and HM are multipliers that adjust the amount to turn
+        HM = 10.0*24/self.downsize_height # based on the current dimensions of self.pixel_matrix.
+        # 'fov' stands for 'field of view'. This is the angle amount
+        # that Cozmo can see to the edges of his camera view.
+        amount_move_head = radians(self.fov_y.radians*((CAMERA_HEIGHT/2)-y*HM)/CAMERA_HEIGHT)
+        amount_rotate = radians(self.fov_x.radians*((CAMERA_WIDTH/2)-x*WM)/CAMERA_WIDTH)
+        if self.moved_too_far_from_center(amount_move_head, amount_rotate):
+            self.state = FOUND_COLOR_STATE
+        if self.state != DRIVING_STATE:
+            self.look_at_it(amount_move_head, amount_rotate)
+        else:
+            self.drive_at_it()
+
+    def moved_too_far_from_center(self, amount_move_head, amount_rotate):
+        ''' Decides whether the center of the blob is too far from the center of Cozmo's view.
+
+            Args:
+                amount_move_head (cozmo.util.Angle): 
+                    the perceived vertical distance of the blob from center-screen
+                amount_rotate (cozmo.util.Angle): 
+                    the perceived horizontal distance of the blob from center-screen
+
+            Returns:
+                too_far (bool): whether the object is too far from center-screen
+        '''
+        too_far_vertical = (amount_move_head.abs_val > self.fov_y/8)
+        too_far_horizontal = (amount_rotate.abs_val > self.fov_x/8)
+        too_far = too_far_vertical or too_far_horizontal
+        return too_far
+
+    def look_at_it(self, amount_move_head, amount_rotate):
+        ''' Calls actions that tilt Cozmo's head and rotate his body toward the color.
+
+            Args:
+               amount_move_head (cozmo.util.Angle): 
+                   the perceived vertical distance of the blob from center-screen
+               amount_rotate (cozmo.util.Angle): 
+                   the perceived horizontal distance of the blob from center-screen           
+        '''
+        self.robot.abort_all_actions()
+        new_head_angle = self.robot.head_angle + amount_move_head
+        self.tilt_head_action = self.robot.set_head_angle(new_head_angle, in_parallel = True)
+        self.rotate_action = self.robot.turn_in_place(amount_rotate, in_parallel = True)
+        if self.state == FOUND_COLOR_STATE:
+            self.amount_turned_recently += amount_move_head.abs_val + amount_rotate.abs_val
+
+    def drive_at_it(self):
```

Comment:
Could be renamed to "drive_to_color_blob" or something more descriptive than "it".